stacks-signer/src/v0/signer_state.rs

```diff
@@ -366,10 +369,22 @@ impl LocalStateMachine {
             }
         };
 
-        // No matter what, if we're in tx replay mode, remove the tx replay set
-        // TODO: in later versions, we will only clear the tx replay
-        // set when replay is completed.
-        prior_state_machine.tx_replay_set = None;
+        if let Some(tx_replay_set) = prior_state_machine.tx_replay_set.clone() {
+            // let replay_set_hash = hash_tx_replay_set(tx_replay_set);
+            let mut hasher = DefaultHasher::new();
+            tx_replay_set.hash(&mut hasher);
+            let replay_set_hash = hasher.finish();
+            if let Ok(true) =
+                db.get_was_block_validated_by_replay_tx(signer_signature_hash, replay_set_hash)
```

Comment:
Hmm, this `if let` kind of scares me even though I know it works. `if let Ok(t) = ..`, for example, would compile, but do something different.

I'd prefer just doing:

```suggestion
            if Ok(true) ==
                db.get_was_block_validated_by_replay_tx(signer_signature_hash, replay_set_hash)
```